### DISCUSSION TITLE
Trouble Forwarding Logs to Standard Output in Pocket Base

### DISCUSSION THREAD

**Discussion by Rfluegel1** [pocketbase/pocketbase#3613] [Q&A] [answered]
Hey everyone,

I’m working on a project where I need to forward logs to standard output using backend hooks in Pocket Base. Below is the script I've pieced together so far:

```javascript
cronAdd("logger", "*/1 * * * *", async () => {
    try {
        const todoSelector = await $app.dao().db().select('*').from('todos').build().rows()
        console.log('todos:', JSON.stringify(todoSelector))
        const _requestsSelector = await $app.logsDao().db().select('*').from('_requests').build().rows()
        console.log('_requests:', JSON.stringify(_requestsSelector))
        const notFoundSelector = await $app.logsDao().db().select('*').from('notFound').build().rows()
        console.log('_requests:', JSON.stringify(notFoundSelector))
    } catch (e) {
        console.log(e)
    }
});
```

When I run this script, here's the output I get:
```
2023/10/26 11:44:24 todos: {"rows":{}}
2023/10/26 11:44:24 _requests: {"rows":{}}
2023/10/26 11:44:24 GoError: SQL logic error: no such table: notFound (1)
```
I’ve already defined and populated the todos table, so I’m puzzled as to why I’m not retrieving any records from it, and I have some 4000 logs according to my pb admin dashboard.

Is there a configuration setting I might have overlooked that enables log forwarding? Or is there something off with my script?

Appreciate any insights or suggestions on this. Thanks!

**Comment by ganigeorgiev** [comment]
I'm not sure I follow.

Note that `JSON.stringify` will print only the public fields of the objects, which doesn't necessary mean that the object is empty.

I'm not sure what `todoSelector` is expected to be in the above, but to retrieve records from the database you can check https://pocketbase.io/docs/js-records/.

If you want to execute raw DB queries see https://pocketbase.io/docs/js-database/#executing-queries.

But what is the desired behavior of the above cron job? Do you really want to print every log record to the stdout?

Also keep in mind that the logs will be refactored in the next released to support all types of logs, not just requests, so the current structure will change.

**Reply by ganigeorgiev** [reply, marked as answer]
If you really want to print every log db write to the stdout you can try to register at the top level `$app.logsDao().afterCreateFunc = (eventDao, model) => { console.log(model) }`, but note that the logs write happen in their own goroutine and the goja callback is not goroutine-safe, aka. you may have to use Go for this (with the logs refactoring there will be an exposed interface to pipe the logs to custom outputs).

**Reply by Rfluegel1** [reply]
> to retrieve records from the database you can check https://pocketbase.io/docs/js-records/.

This seems to be what I was looking for and works for me when I set the filter to true=true.

> to print every log db write to the stdout you can try to register at the top level $app.logsDao().afterCreateFunc = (eventDao, model) => { console.log(model) }

This is the big brain answer that didn't quite click for me working on my own.

I consider this post answered

---
As for why I am interested in printing logs to standard output, fly.io recommends making use of betterstack for shipping logs. To do so pocketbase _requests must be pushed to application logs.

**Reply by Rfluegel1** [reply]
Also your response time is insanely fast. Praise be
